src/widgets/hierarchical-menu/hierarchical-menu.js

```diff
@@ -137,19 +135,23 @@ function hierarchicalMenu({
             subValue.data = this._prepareFacetValues(subValue.data, state);
           }
 
-          subValue.url = this._createURL(state, subValue);
-
           return subValue;
         });
     },
-    render: function({results, state}) {
+    render: function({results, state, createURL}) {
       let facetValues = results.getFacetValues(hierarchicalFacetName, {sortBy: sortBy}).data || [];
       facetValues = this._prepareFacetValues(facetValues, state);
 
+      // Bind createURL to this specific attribute
+      function _createURL(facetValue) {
+        return createURL(state.toggleRefinement(hierarchicalFacetName, facetValue));
+      }
+
       ReactDOM.render(
         <RefinementList
           attributeNameKey="path"
           collapsible={collapsible}
+          createURL={_createURL}
```

Comment:
The only "better" optimisation we could do is know upfront that urlSync is deactivated (thus, no different urls per facet values) but I do not see an easy way to do so.

And it may be an edge case we do not need to support.